cpp/src/arrow/array/data.cc

```diff
@@ -236,6 +236,19 @@ void SetOffsetsForScalar(ArraySpan* span, offset_type* buffer, int64_t value_siz
   span->buffers[buffer_index].size = 2 * sizeof(offset_type);
 }
 
+template <typename ArrowType>
+void SetRunForScalar(ArraySpan* span, std::shared_ptr<DataType> type,
+                     uint64_t* scratch_space) {
+  // Create a lenght-1 array with the value 1 for a REE scalar
+  using run_type = typename ArrowType::c_type;
```

Comment:
`run_end_type` (run_type is ambiguous -- could be the type of the value in the run).